### DISCUSSION TITLE
Get the stdout, stderr and status of each shell.send(cmd) ?

### DISCUSSION THREAD

**Discussion by jipen** [JacobCallahan/Hussh#18] [Q&A] [answered]
Hi,

Is there a way to keep the stdout, stderr and status when a single shell.send() is executed ?

Thx.

**Comment by JacobCallahan** [comment, marked as answer]
Unfortunately, this isn't currently possible due to the way we have to read from the channel. Basically, we need to tell it we're done before we can get the information back from it. If you're curious the implementation is here: https://github.com/JacobCallahan/Hussh/blob/master/src/connection.rs#L559

This may be possible in the future when I add in the AsyncConnection class I'm thinking about..

If you, or anyone else, does know of a good way to do this then please do let me know (or raise a PR)!

**Reply by jipen** [reply]
I don't know Rust. For the moment...😉

**Reply by JacobCallahan** [reply]
Not a problem at all, you shouldn't have to in order to be a user of this project.

**Reply by jipen** [reply]
But it could help to read and understand your sources !  😉

**Comment by JacobCallahan** [comment]
@jipen I've just added async functionality today that does just this! However I will say that my initial testing showed that the status was always reported as 0 when pulled within an interactive session so not sure if that's something that can be resolved. But if you want to test it yourself feel free to
